GAD 7 - Radixsort/tests/radix/RadixSortTester.java

```diff
@@ -19,845 +20,997 @@ void printLine() {
-        assertArrayEquals(new int[]{4, 9, 6, 6, 8, 8, 5, 2, 6, 8, 7, 4, 7, 1, 4, 6, 2, 8, 2, 3, 3}, elements);
 
-        buckets = new ArrayList[8];
-        for (int i = 0; i < 8; i++) {
-            buckets[i] = new ArrayList<>();
+        interface ArrayGenerator {
+            int[] generate();
         }
-        {
-            buckets[0].addAll(Arrays.stream(new int[] {4, 3}).boxed().toList());
-            buckets[1].addAll(Arrays.stream(new int[] {7, 4}).boxed().toList());
-            buckets[2].addAll(Arrays.stream(new int[] {5, 0}).boxed().toList());
-            buckets[3].addAll(Arrays.stream(new int[] {0, 0}).boxed().toList());
-            buckets[4].addAll(Arrays.stream(new int[] {9, 7}).boxed().toList());
-            buckets[5].addAll(Arrays.stream(new int[] {4, 0}).boxed().toList());
-            buckets[6].addAll(Arrays.stream(new int[] {0, 8}).boxed().toList());
-            buckets[7].addAll(Arrays.stream(new int[] {1, 4}).boxed().toList());
-        }
-        elements = new int[] {4, 3, 7, 4, 5, 0, 0, 0, 9, 7, 4, 0, 0, 8, 1, 4};
-        RadixSort.concatenate(buckets, elements);
-        assertArrayEquals(new int[]{4, 3, 7, 4, 5, 0, 0, 0, 9, 7, 4, 0, 0, 8, 1, 4}, elements);
     }
 
+
     /**
-     * Tests to see if your ksort is sorting the respective decimal place correctly or not.
      * @author Aamin
      */
-    @Test
-    public void ksortTests() {
-        int[] elements;
-        elements = new int[]{0};
-        RadixSort.kSort(elements, 1);
-        assertArrayEquals(new int[] {0}, elements);
-
-        elements = new int[]{0, 1, 1};
-        RadixSort.kSort(elements, 0);
-        assertArrayEquals(new int[] {0, 1, 1}, elements);
-
-        elements = new int[]{2, 3, 3, 0, 0};
-        RadixSort.kSort(elements, 1);
-        assertArrayEquals(new int[] {2, 3, 3, 0, 0}, elements);
-
-        elements = new int[]{5, 2, 4, 3, 3, 3, 7};
-        RadixSort.kSort(elements, 0);
-        assertArrayEquals(new int[] {2, 3, 3, 3, 4, 5, 7}, elements);
-
-        elements = new int[]{23, 14, 12, 21, 22, 2, 0, 6, 10};
-        RadixSort.kSort(elements, 1);
-        assertArrayEquals(new int[] {2, 0, 6, 14, 12, 10, 23, 21, 22}, elements);
-
-        elements = new int[]{24, 4, 1, 1, 70, 29, 8, 8, 18, 33, 36};
-        RadixSort.kSort(elements, 0);
-        assertArrayEquals(new int[] {70, 1, 1, 33, 24, 4, 36, 8, 8, 18, 29}, elements);
-
-        elements = new int[]{20, 130, 153, 76, 138, 121, 188, 15, 51, 59, 168, 3, 162};
-        RadixSort.kSort(elements, 4);
-        assertArrayEquals(new int[] {20, 130, 153, 76, 138, 121, 188, 15, 51, 59, 168, 3, 162}, elements);
-
-        elements = new int[]{38, 20, 26, 62, 82, 8, 451, 139, 472, 440, 397, 387, 325, 69, 231};
-        RadixSort.kSort(elements, 3);
-        assertArrayEquals(new int[] {38, 20, 26, 62, 82, 8, 451, 139, 472, 440, 397, 387, 325, 69, 231}, elements);
+    @Nested
+    @DisplayName("Aamin's Tests")
+    class aaminsTests {
```

Comment:
> Class names should be nouns, in mixed case with the first letter of each internal word capitalized

https://www.oracle.com/java/technologies/javase/codeconventions-namingconventions.html

Plus in general I find irrespective of the scenario that names should be capitalised.
Other than that! Everything else works like a charm, better than before

Reply:
That was the pruduct of late night frutration fair point this should be capitalized.

Reply:
Haha, happens to the best of us. You still did an amazing job, especially considering that you did it in the middle of night

Reply:
Most of it was done before 10 pm :)